hibernate-core/src/main/java/org/hibernate/metamodel/mapping/NaturalIdMapping.java

```diff
@@ -39,6 +40,7 @@
  *
  * @author Steve Ebersole
  */
+@Incubating
 public interface NaturalIdMapping extends VirtualModelPart {
```

Comment:
I have removed the session from the methods of this SPI, since it is not needed, and because I needed to call `extractNaturalIdFromEntityState()` from a place where I did not have one.